core/deployment/src/main/java/io/quarkus/deployment/dev/RuntimeUpdatesProcessor.java

```diff
@@ -389,13 +534,14 @@ public Throwable getCompileProblem() {
     }
 
     private void checkForClassFilesChangesInModule(DevModeContext.ModuleInfo module, List<Path> moduleChangedSourceFiles,
-            boolean isInitialRun, ClassScanResult classScanResult) {
-        if (module.getClassesPath() == null) {
+            boolean isInitialRun, ClassScanResult classScanResult,
+            Function<DevModeContext.ModuleInfo, DevModeContext.CompilationUnit> cuf, TimestampSet timestampSet) {
+        if (cuf.apply(module).getClassesPath() == null) {
             return;
         }
 
         try {
-            for (String folder : module.getClassesPath().split(File.pathSeparator)) {
+            for (String folder : cuf.apply(module).getClassesPath().split(File.pathSeparator)) {
```

Comment:
Is `cuf` being applied twice to the same `module` on purpose (line 539 and 544)?

Reply:
Looks like it's not a big deal.